server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -271,6 +271,13 @@ public void deserialize() throws IOException {
         }
       }
     }
+
+    // upgrade from v0.12 to v0.13, we need to rewrite the TsFileResource if the previous time index
+    // is file time index
+    if (timeIndexType == 0) {
```

Comment:
Maybe you can also move the serialize() logic into deserialize of V012FileTimeIndex so that all update related logics are put together.